src/state/digital-exchange/actions.js

```diff
@@ -6,6 +6,16 @@ import { setSelectedDEExtraFilter } from 'state/digital-exchange/extra-filters/a
 import { ALL_CATEGORIES_CATEGORY } from 'state/digital-exchange/categories/const';
 import { DE_COMPONENTS_EXTRA_FILTERS } from 'state/digital-exchange/extra-filters/const';
 import { fetchDEComponents, setDEFilter } from 'state/digital-exchange/components/actions';
+import { getSelectedDEExtraFilter } from './extra-filters/selectors';
+
+const genFilterParams = (filter, getState) => {
+  const filters = getDEFilters(getState());
+  const selectedExtraFilter = getSelectedDEExtraFilter(getState());
+  const selectedExtraFilterParams = selectedExtraFilter ? convertToQueryString(DE_COMPONENTS_EXTRA_FILTERS[selectedExtraFilter]) : '';
+  const filterParams = filters[filter] ? convertToQueryString(filters[filter]) : '';
+  const params = `?${selectedExtraFilterParams.slice(1)}&${filterParams.slice(1)}`;
```

Comment:
when `filterParams` return as blank '', this might leave out '&' at the end of params. Can you make sure to avoid this case?